src/bin/solve.rs

```diff
@@ -51,14 +52,68 @@ fn generate_fwopcache<P: AsRef<Path>>(path: P) -> io::Result<File> {
             })
             .or_insert_with(|| (i, i.count_ones()));
     }
-    for (k, v) in cache.entries() {
-        file.write(k.to_ne_bytes())?;
-        file.write(&v.to_ne_bytes())?;
+
+    for (k, v) in &cache {
+        file.write(&k.to_ne_bytes())?;
+        file.write(&v.0.to_ne_bytes())?;
     }
     file.flush()?;
     Ok(file)
 }
 
+// fn get_entry_from_cache<F: Read, F:Seek>(cache: F) -> Option<u64> {
```

Comment:
Try to make clean commits, without extra junk in them. Adding by hunk is a good way to avoid this.